### PR TITLE
pythonPackages.poppler-qt5: 0.24.2 -> 21.1.0

### PR DESCRIPTION
**Pull request by eduardosm**
Builds using sip 5

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by eduardosm** [comment]
Result of `nixpkgs-review pr 120798` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>frescobaldi</li>
    <li>krop</li>
    <li>python38Packages.poppler-qt5</li>
    <li>python39Packages.poppler-qt5</li>
  </ul>
</details>

File: pkgs/development/python-modules/poppler-qt5/default.nix [hunk 1]
```diff
@@ -1,35 +1,78 @@
...
+, poppler
+, pkg-config
+, fetchpatch
 , substituteAll
 }:
 
 buildPythonPackage rec {
-  pname = "python-poppler-qt5";
-  version = "0.24.2";
+  pname = "poppler-qt5";
```
Comment (mweinelt) [inline pkgs/development/python-modules/poppler-qt5/default.nix:19]: https://github.com/frescobaldi/python-poppler-qt5/blob/0c50dd05fdab563a0c9decc2ed784712f37c822f/setup.py#L4

python-poppler-qt5 is the actual library name.

Comment (eduardosm) [reply pkgs/development/python-modules/poppler-qt5/default.nix]: Reverted... I had removed it because it looked weird with the double python prefix.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120798` at 9a102640 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>frescobaldi</li>
    <li>krop</li>
    <li>python38Packages.poppler-qt5</li>
    <li>python39Packages.poppler-qt5</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/poppler-qt5/default.nix:20:0:
```
   |
20 |   version = "21.1.0";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/poppler-qt5/default.nix:84:5:
```
   |
84 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/poppler-qt5/default.nix:83:5:
```
   |
83 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `substituteAll`.
Near pkgs/development/python-modules/poppler-qt5/default.nix:15:3:
```
   |
15 | , substituteAll
   |   ^
```
</li>
  </ul>
</details>

**Comment by eduardosm** [comment]
After adding `pythonImportsCheck = [ "popplerqt5" ];`:

```
Check whether the following modules can be imported: popplerqt5
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 1, in <lambda>
  File "/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'popplerqt5'
```

File: pkgs/development/python-modules/poppler-qt5/default.nix [hunk 2]
```diff
@@ -1,44 +1,87 @@
...
   patches = [
-    (substituteAll {
-      src = ./poppler-include-dir.patch;
-      poppler_include_dir = "${poppler.dev}/include/poppler";
-    })
+    # Fix for https://github.com/frescobaldi/python-poppler-qt5/issues/43
     (fetchpatch {
-      url = "https://github.com/wbsoft/python-poppler-qt5/commit/faf4d1308f89560b0d849671226e3080dfc72e79.patch";
-      sha256 = "18krhh6wzsnpxzlzv02nginb1vralla8ai24zqk10nc4mj6fkj86";
+      url = "https://patch-diff.githubusercontent.com/raw/frescobaldi/python-poppler-qt5/pull/45.patch";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/poppler-qt5/default.nix:31]: Please fetch the commit(s) instead of the PR.

Comment (eduardosm) [reply pkgs/development/python-modules/poppler-qt5/default.nix]: Fixed

**Comment by eduardosm** [comment]
I could not figure out why `pythonImportsCheck` is failing, so I implemented it manually.

**Comment by SuperSandro2000** [comment]
> I could not figure out why `pythonImportsCheck` is failing, so I implemented it manually.

How did you get it working?

**Comment by eduardosm** [comment]
> How did you get it working?

Instead of manually implementing `buildPhase` and `installPhase`, I set `format = "pyproject"`, whose builder automatically calls sip 5. After that, `pythonImportsCheck` worked again. I still couldn't figure out why it didn't work before.

**Comment by eduardosm** [comment]
Rebased to integrate with https://github.com/NixOS/nixpkgs/pull/121036.

**Pull request merged**